main.ts

```diff
@@ -139,86 +114,106 @@ export default class SmarterMDhotkeys extends Plugin {
+		if (selected.includes(" ")) {
+			const firstWordRange = WordUnderCursor(selStart);
 			let lastWordRange;
 
 			//findatword reads to the right, so without "-1" the space would be read
-			if (selected.match(/[.,;:\-–—]$/) == null){
+			if (selected.match(/[.,;:\-–—]$/) == null) {
 				selEnd.ch--;
-				lastWordRange = WordUnderCursor (selEnd);
+				lastWordRange = WordUnderCursor(selEnd);
 				selEnd.ch++;
 			} else {
-				lastWordRange = WordUnderCursor (selEnd);
+				lastWordRange = WordUnderCursor(selEnd);
 			}
+			const { anchor, head } = lastWordRange;
 
 			// Fix for punctuation messing up selection
-			const lastWord = editor.getRange(lastWordRange.anchor, lastWordRange.head);
-			if (lastWord.match(/^[.,;:\-–—]/) != null){
-				lastWordRange.head.ch = lastWordRange.anchor.ch + 1;
+			const lastWord = editor.getRange(anchor, head);
+			// I think it's faster to `test` a regex than look for `match` if you only want to see if it matches
+			if (/^[.,;:\-–—]/.test(lastWord)) {
+				head.ch = anchor.ch + 1;
 			}
 
-			editor.setSelection(firstWordRange.anchor, lastWordRange.head);
+			editor.setSelection(firstWordRange.anchor, head);
 			multiWordExpanded = true;
 		}
 
 		// Get properties of selection
 		const selectedText = editor.getSelection();
-		const sp = editor.posToOffset(editor.getCursor("from")); // Starting position
+		// 'o' instead of 'p' because it's an offset, not a position
+		const so = editor.posToOffset(editor.getCursor("from")); // Starting offset
 		const len = selectedText.length;
-		const charsBefore = editor.getRange(offToPos(sp - blen), offToPos(sp));
-		const charsAfter = editor.getRange(offToPos(sp + len), offToPos(sp + len + alen));
-		const firstChars = editor.getRange(offToPos(sp), offToPos(sp + blen));
-		const lastChars = editor.getRange(offToPos(sp + len - alen), offToPos(sp + len));
+		const eo = so + len; // Ending offset
+
+		const [charsBefore, charsAfter] = [
+			editor.getRange(offToPos(so - blen), offToPos(so)),
+			editor.getRange(offToPos(eo), offToPos(eo + alen)),
+		];
+
+		const [firstChars, lastChars] = [
+			editor.getRange(offToPos(so), offToPos(so + blen)),
+			editor.getRange(offToPos(eo - alen), offToPos(eo)),
+		];
 
 		// Undo surrounding markup (outside selection)
-		if (charsBefore == beforeStr && charsAfter == afterStr) {
-			editor.setSelection(offToPos(sp - blen), offToPos(sp + len + alen));
+		if (charsBefore === beforeStr && charsAfter === afterStr) {
+			editor.setSelection(offToPos(so - blen), offToPos(eo + alen));
 			editor.replaceSelection(selectedText);
 			if (wordExpanded) {
+				// I've had issues when mutating properties of a `const` position before... I recommend making a copy of the position to use for this purpose only
 				noSelPosition.ch -= blen;
 				editor.setCursor(noSelPosition);
-			} else if (multiWordExpanded){
+			} else if (multiWordExpanded) {
+				// Same here. You might be introducing unintended side effects by mutating this object

```

Comment:
care to elaborate?

Reply:
For example, if you were to use `noSelPosition` somewhere else, you would rightfully assume that it is the same as when you first defined it.
But just because it's a const, doesn't mean you can mutate properties _inside_ the const object.
Therefore, it isn't actually the same object anymore